app/templates/_base_page.html

```diff
@@ -4,6 +4,26 @@
 
 {% from "components/breadcrumbs/macro.njk" import govukBreadcrumbs %}
 {% from "components/button/macro.njk" import govukButton %}
+{% from "components/error-summary/macro.njk" import govukErrorSummary %}
+
+{% macro dmErrorSummary(params) %}
+  {% if params.errors %}
+    {% set locals = namespace(errorList = []) %}
+    {% for error in params.errors.values() %}
+      {% set locals.errorList = locals.errorList + [{
+        'text': error.message,
+        'href': '#input-' + error.input_name
+      }] %}
+
+    {% endfor %}
```

Comment:
If this could be done in the backend and returned to the view as part of `errors` it would help simplify the code and we call `govukErrorSummary` like this:

```
{% if errors %}
  {{ govukErrorSummary({
       "title": "There is a problem",
       "errorList": errors.errorList})
  }}
{% endif %}
```

Reply:
Let's see what we can do!